packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -100,8 +100,19 @@ class CozyRealtime {
    */
   async connect() {
     logger.info('connecting…')
-    await this.retryManager.waitBeforeNextAttempt()
-    this.createSocket()
+    // avoid multiple concurrent calls to connect, keeps the first one
+    if (!this.waitingForConnect) {
```

Comment:
Nit: I find it clearer when conditions are checked positively, and early bail out is used.

```
if (this.waitingForConnect) {
  logger.debug('Skipping reconnect because one is already waiting')
  return
}


... normal flow of function ...